### ISSUE TITLE
Extention no longer works: New Notification: 32bit Java not supported: I run 64bit version.

### ISSUE THREAD

**Issue by dastagg** [IBM/zopeneditor-about#57]
Recently, a new Notification Popup is being displayed when you retrieve a member from a PDS:

"A 32 bit Java Runtime is not supported for the IBM Z Open Editor extension. Please follow our documentation to configure a 64 bit version of Java 8 or newer."

The Extension no longer works at all. 
No spell checking,
No suggestions,
No "Intelligence",
No Nothing.

## Development environment used

- Editor Platform
  - [X ] Visual Studio Code
  - [ ] CodeReady Workspaces
  - [ ] Eclipse Che
  - [ ] Standlone Theia
- Platform Version: OSS
- Operating System (on which VS Code runs such as Windows 10 2004, otherwise name and version of platform such as OpenShift v4.3): Arco Linux (Arch Linux) 64bit.
- Java Version (when using VS Code or Theia, extecute `java -version` and paste the details here):
openjdk version "14.0.1" 2020-04-14
OpenJDK Runtime Environment (build 14.0.1+7)
OpenJDK 64-Bit Server VM (build 14.0.1+7, mixed mode)

- Logs attached (see [here](https://ibm.github.io/zopeneditor-about/Docs/knownissues.html) how to get them): yes/no

## Problem Description

Detailed steps for reproducing the problem:

1. First step

Observed behavior:
Open VSCode
Click on Zowe Explorer
Click on Profile
Filter by Zid
Expand the CBL pds
Click on member to display the source code.
Notification Popup in lower right hand corner:
"A 32 bit Java Runtime is not supported for the IBM Z Open Editor extension. Please follow our documentation to configure a 64 bit version of Java 8 or newer."
with a button "Configure Java" which when clicked on takes you [to:](https://ibm.github.io/zopeneditor-about/Docs/getting_started.html#verifying-the-installation-of-ibm-z-open-editor)

I don't have (and haven't for a long time) a 32bit version of Java running on my computer.
I'm not sure why a message like this would be displayed.
AND Nothing works.
It worked fine last week and now it's like the Extension isn't even there.

-

Designed behavior:

**Comment by edrubins** [comment]
Same problem.

Windows 10 Version 2002.
Java SDK: OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.7+10, mixed mode)
VS Code Version 1.46.0 (problem surfaced while running version 1.45.1.)
Performing a complete uninstall, including the .vscode and .zowe directories, and reinstall didn't solve the problem.
Running VS Code version 1.45.1 on Ubuntu 16.04 with no problem.

**Comment by lauren-li** [comment]
@dastagg @edrubins Thank you for reporting this issue. Can both of you please share what version of IBM Z Open Editor you are experiencing the problem with?

@edrubins Were you previously able to have IBM Z Open Editor running on your Windows 10 Version 2002 system's VS Code?

Our team will investigate further and look into recreating the issue. Thanks again!

**Comment by kdbeall** [comment]
@edrubins I've managed to reproduce the issue.

**Comment by dastagg** [comment]
I just went to check. It was 0.5.0. When I clicked on Extensions, it had a "installing" box. It upgraded itself to 0.5.1. I closed/re-opened VSCode and the same issue is still present.

**Comment by dastagg** [comment]
Here is more info about VSCode:
Version: 1.46.0
Commit: a5d1cc28bb5da32ec67e86cc50f84c67cc690321
Date: 2020-06-11T14:57:41.218Z
Electron: 7.1.14
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Linux x64 5.7.2-arch1-1

**Comment by dastagg** [comment]
And a little more info:
$ archlinux-java status                  
Available Java environments:
  java-11-openjdk (default)
  java-14-openjdk

I tried switching to 14 and got the same issue.

**Comment by edrubins** [comment]
> @dastagg @edrubins Thank you for reporting this issue. Can both of you please share what version of IBM Z Open Editor you are experiencing the problem with?
> 
> @edrubins Were you previously able to have IBM Z Open Editor running on your Windows 10 Version 2002 system's VS Code?
> 
> Our team will investigate further and look into recreating the issue. Thanks again!

Z Open Editor extension is 0.5.0. The problem started suddenly some time last Thursday. It was working fine until then.

**Comment by kdbeall** [comment]
@dastagg  @edrubins,

We've found the root cause of the issue and are working to get this fixed.
Thank you for bringing this to our attention!

**Comment by edrubins** [comment]
It took a while to find a JDK that worked. The IBM Version 8 packaged with Eclipse is 32 bit. Oracle's 64 bit version 8 is installed in C:\Prgograms Files\ (x86), which made it  imposable, for me at least, to make the editor happy with the path.

The one JDK that finally worked is AdoptOpenJDK  with the Openj9 Hotspot.

**Comment by dastagg** [comment]
Okay, I installed jdk8-openjdk and I can report that the functionality has returned.
My question now is: Why do I have to go "backwards"?
Last week jdk14-openjdk (or jdk11-openjdk) worked, so what changed?

**Comment by dastagg** [comment]
Okay, fine, it appears no one will answer that question but the issue has been "fixed" so I'll close the issue.

**Comment by dastagg** [comment]
I'm re-opening this issue just to make sure this is visible. If someone else (on the team) wants to close it, that's fine.

There are a few of us that are participating in the Open Mainframe Project/COBOL using VSCode class that have been running into this same issue.

One of the participants - Joao Carvalho - was determined to figure out what was causing the problem.

First, let's start with the text from the README of the extension in VSCode:
```
Java SDK version 8 or later - 64 bit: The COBOL and PL/I language servers included in this extension are implemented in Java. Therefore, you need to install and configure a 64-bit Java SDK in order to start it successfully.
  You can use Oracle Java SDK 8 or the LTS version 11 of Oracle Java. Alternatively, you can use version 8 or 11 of the OpenJDK.
  
  Note, in all cases, you need to download and configure a full JDK and not just a Client Runtime version of Java. The language servers will not work with only a Client Runtime version.
```

This text was correct for version 0.4.9 but through checking, determined that something changed in version 0.5.0.

On Windows:
Java 8 Runtime - Works v0.4.9, v0.5.0.
Java 8 JDK - Works v0.4.9, v.0.5.0.
Java 11 JDK - Works v0.4.9 but does not work v0.5.0.
Java 14 JDK - Works v0.4.9 but does not work v0.5.0.

I have also confirmed (using Arch Linux) that jdk8-openjdk works but jdk11-openjdk and jdk14-openjdk do not with v0.5.0 or v0.5.1. I have not confirmed v4.9.0 but it worked about 2 weeks ago which I will guess was v0.4.9 and then was upgraded to v0.5.0 and stopped working.

So, something changed in v0.5.0.

Now, that's one thing but there are a number of people that are joining this project to learn how to use VSCode to program on the Mainframe using JCL and COBOL.
Based on the text in the README, they are running into problems because they are installing v0.5.0 (maybe v0.5.1) of the Editor Extension and then based on that, either the 11 or 14 JDK.

So, depending on the state of the "Oops, we need to fix this in the Editor Extension code" status, you should consider updating the text in the README so people aren't installing a Java program (or at least a version of one) that doesn't work.

At least until the Editor Extension code is fixed and a more recent version of the JDK will work correctly.

**Comment by kdbeall** [comment]
@dastagg Thank you for reopening the issue. I have been on vacation. Being very general, there was use of a flag which was deprecated in higher versions of Java. The fix was fairly trivial, and we hope to have it out soon.

**Comment by phaumer** [comment]
This issue was fixed in 0.5.2. @dastagg and @edrubins could you try this again. 
We tested the fix with Java SDKs from OpenJDK (HS and J9) v8, 11, 14 and Oracle v11 and 14 and IBM JDK 8.

**Comment by dastagg** [comment]
I have tested it with java-11-openjdk and java-14-openjdk on Arch Linux and it is working correctly now.
Thank You for fixing it.
